docs/use_cases/embeddings_on_browser.md

```diff
@@ -0,0 +1,494 @@
+
+Though our Encoder's sentence embeddings are pre-trained, they can also be fine-tuned for specific tasks, even when there isn't much task-specific training data. To ensure that the encoder is a versatile tool supporting multiple downstream tasks, it can be trained using multi-task learning.
+
+
+Okay, let's get started. 
+(Unless otherwise indicated, we'll be using Typescript.)
+
+## Our step-by-step tutorial
+
+### Import modules
+```tsx
+import React, { FC, useState, useEffect, useCallback } from 'react';
+import {
+  Box,
+  Grid,
+  Typography,
+  TextField,
+  Paper,
+  Button,
+  CircularProgress,
+} from "@mui/material";
+
+import '@tensorflow/tfjs-backend-cpu';
+import '@tensorflow/tfjs-backend-webgl';
+
+import * as use from '@tensorflow-models/universal-sentence-encoder';
+import * as tf from '@tensorflow/tfjs-core';
+import { interpolateGreens } from 'd3-scale-chromatic';
+```
+
+### State variables to manage user input, loading state, and results
+
+We use the **`useState`** hook in a React functional component to manage user input, track loading states of machine learning models, and update the user interface with the results, including the similarity matrix visualization.
+
+```tsx
+// State variables to manage user input, loading state, and results
+
+// sentences - stores user input 
+  const [sentences, setSentences] = useState<string>('');
+
+// sentencesList - stores sentences split into array
+  const [sentencesList, setSentencesList] = useState<string[]>([]);
+
+// embeddings - stores generated embeddings
+  const [embeddings, setEmbeddings] = useState<string>('');
+
+// modelLoading - tracks model loading state
+  const [modelLoading, setModelLoading] = useState(false);
+
+// modelComputing - tracks model computing state
+  const [modelComputing, setModelComputing] = useState(false);
+
+// showSimilarityMatrix - controls matrix visibility
+  const [showSimilarityMatrix, setShowSimilarityMatrix] = useState(false);
+
+// embeddingModel - stores sentence embeddings  
+  const [embeddingModel, setEmbeddingModel] = useState<tf.Tensor2D>(tf.tensor2d([[1, 2], [3, 4]]));
+
+// canvasSize - size of canvas for matrix
+  const [canvasSize, setCanvasSize] = useState(0);
+
+// similarityMatrix - stores similarity matrix
+  const [similarityMatrix, setSimilarityMatrix] = useState<number[][] | null>(null);
+```
+
+### Function to toggle the display of the similarity matrix
+
+The **`handleSimilarityMatrix`** function is called in response to user input, toggling the display of a UI similarity matrix - by changing the **`showSimilarityMatrix`** state variable. If the matrix was previously shown, the **`handleSimilarityMatrix`** hides it by setting it to **`null`**. If the matrix wasn't shown, the **`handleSimilarityMatrix`** calculates the matrix and sets it to display in the UI.
+
+```tsx
```

Comment:
line 134
why use three ticks and tsx instead of three ticks and js to indicate javascript?